src/functions/function-command/build-and-deploy-function.ts

```diff
@@ -169,28 +158,49 @@ export async function buildFunction(context?: FunctionNode): Promise<CliExitData
   }
 }
 
-export async function deployFunction(context?: FunctionNode): Promise<void> {
-  const selectedFolderPick: FolderPick = await selectedFolder(context);
-  if (!selectedFolderPick && !context) {
+export async function deployFunction(context?: FunctionNode): Promise<CliExitData> {
+  if (!context) {
     return null;
   }
-  const funcData = await functionImage(
-    context ? context.contextPath : selectedFolderPick.workspaceFolder.uri,
-    true,
-    context.getName(),
-    context?.getParent()?.getName(),
-  );
+  const funcYaml: string = await fs.readFile(path.join(context.contextPath.fsPath, 'func.yaml'), 'utf-8');
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  const getFuncYaml = yaml.safeLoadAll(funcYaml);
+  if (!getFuncYaml?.[0]?.image) {
+    const response: string = await vscode.window.showInformationMessage(
+      'Image not found in func.yaml. Do you want to build before deploy?',
+      'Build',
+      'Cancel',
+    );
+    if (response === 'Build') {
+      await buildFunction(context);
+    } else {
+      return null;
+    }
+  }
+  const funcData = await functionImage(context.contextPath, true, context.getName(), context?.getParent()?.getName());
   if (!funcData) {
     return null;
   }
   telemetryLog('function_deploy_command', 'Deploy command execute');
   // eslint-disable-next-line @typescript-eslint/no-floating-promises
   functionExplorer.refresh();
-  await knExecutor.executeInTerminal(
-    await FuncAPI.deployFunc(
-      context ? context.contextPath.fsPath : selectedFolderPick.workspaceFolder.uri.fsPath,
-      funcData.image,
-      context?.getParent()?.getName(),
-    ),
+  const command = await FuncAPI.deployFunc(context.contextPath.fsPath, funcData.image, context?.getParent()?.getName());
+  const name = `Deploy: ${context.getName()}`;
+  if (!STILL_EXECUTING_COMMAND.get(name)) {
+    const result = await executeCommandInOutputChannels(command, name);
+    if (restartDeployCommand.get(context.getName())) {
```

Comment:
Always the same comment as i find this way of writing code too chaotic. For me the processes should be handled by some external manager/controller. 

Or a simplified way easier to follow
```
executing_processes = new Map<string, process>();

function deployFunction(context?: FunctionNode): Promise<CliExitData> {
     if command still executing {
         const status = await vscode.window.showWarningMessage(
              `The Function ${command.cliArguments[0]}: ${context.getName()} is already active.`,
               'Restart',
            );
        if (status === 'Restart') {
              executing_processes.get(name).kill
        } else {
           return
        }
     }

      code to execute a command 




```

Reply:
@sudhirverma Let's fix the review in the same PR. Is it a lengthy task or can be done within this PR only ?

Reply:
Here I have included the `simplified way easier to follow` that Luca had suggested. I am thinking to create a new PR which will handle the `processes by some external manager/controller.`